src/main.ts

```diff
@@ -235,57 +274,71 @@ export class MagentoStack extends Stack {
       ? this.node.tryGetContext('os_master_user_name')
       : 'magento-os-master';
 
-    const osDomain = new opensearch.Domain(this, 'Domain', {
-      version: opensearch.EngineVersion.OPENSEARCH_1_0,
-      domainName: OS_DOMAIN,
-      //accessPolicies: [osPolicy], // Default No access policies for magento
-      removalPolicy: RemovalPolicy.DESTROY,
-      securityGroups: [openSearchSG],
-      //If you want more capacity for Opensearch . default 1 instance r5.large.search datanode; no dedicated master nodes
-      // capacity: {
-      //   masterNodes: 5,
-      //   dataNodes: 20,
-      // },
-      ebs: {
-        volumeSize: 20,
-      },
-      //if you need, else only 1 az
-      // zoneAwareness: {
-      //   availabilityZoneCount: 3,
-      // },
-      logging: {
-        slowSearchLogEnabled: true,
-        appLogEnabled: true,
-        slowIndexLogEnabled: true,
-      },
+    const OS_DOMAIN_ENDPOINT = this.node.tryGetContext('os_domain_endpoint')
+      ? this.node.tryGetContext('os_domain_endpoint')
+      : undefined;
 
-      //encryption
-      enforceHttps: true,
-      nodeToNodeEncryption: true,
-      encryptionAtRest: {
-        enabled: true,
-      },
-      fineGrainedAccessControl: {
-        masterUserName: OS_MASTER_USER_NAME,
-        masterUserPassword: magentoOpensearchAdminPassword.secretValue,
-      },
-      useUnsignedBasicAuth: true,
-      enableVersionUpgrade: true,
-    });
+    var osDomain;
+    if (OS_DOMAIN_ENDPOINT) {
+      // If we uses an existing OpenSearch Domain
+      osDomain = opensearch.Domain.fromDomainEndpoint(this, 'domainImport', OS_DOMAIN_ENDPOINT);
+    } else {
+      osDomain = new opensearch.Domain(this, 'Domain', {
+        version: opensearch.EngineVersion.OPENSEARCH_1_0,
+        domainName: OS_DOMAIN,
```

Comment:
TODO: We need to make opensearch to deployed within the VPC. Currently The default settings opens the Opensearch endpoint to Internet.